### PR TITLE
Various build improvements

### PR DESCRIPTION
**Pull request by benesch**
See commit messages.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/14032)

### CODE REVIEW

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[build/cockroach.rb, line 8 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14032#-KeoF2SfJnwQtrdAVQOm:-KeoF2SfJnwQtrdAVQOn:b-6dni44) ([raw file](https://github.com/cockroachdb/cockroach/blob/a95f6e6c292746421acd51c66d8a09c761cc46da/build/cockroach.rb#L8)):*
> ```Ruby
>   url "https://github.com/cockroachdb/cockroach.git",
>       :tag => "beta-20170223",
>       :revision => "4c8949513d287b92be1c167615398a2a53fe39da"
> ```

i think there was a reason for this. care to dig through the git log?

cc @mberhault 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14032#-:-KeoFDoeSfNgrSj_floM:bnfp4nl)*

File: build/cockroach.rb [hunk 1]
```diff
@@ -1,11 +1,8 @@
-require "language/go"
-
 class Cockroach < Formula
   desc "Distributed SQL database"
   homepage "https://www.cockroachlabs.com"
   url "https://github.com/cockroachdb/cockroach.git",
-      :tag => "beta-20170223",
-      :revision => "4c8949513d287b92be1c167615398a2a53fe39da"
+      :tag => "beta-20170223"
```
Comment (mberhault) [inline build/cockroach.rb]: does this work?

**Comment by mberhault** [comment]
dunno, I have no little to no experience with homebrew. I've never tried to install cockroach through it

**Comment by benesch** [comment]
@tamird @mberhault specifying both `tag` and `revision` dates back to Peter's commit to add the Homebrew recipe, so I assume it's just historical happenstance. Homebrew is perfectly happy with just a `tag`; it uses the revision, if specified alongside a tag, to verify the revision pointed to by the tag is what the formula expects.

This functionality exists to prevent upstream Git committers from force-pushing a tag and changing the code a core Homebrew formula points at; since we control both the repository and the formula, I don't think we need the extra verification. If our tags are force-pushed we've got other problems. Plus I'm hoping to eventually switch over to tarball sources per #14031.

**Pull request merged**